back/app/models/initiative.rb

```diff
@@ -58,8 +61,8 @@ class Initiative < ApplicationRecord
   with_options unless: :draft? do |post|
     post.validates :title_multiloc, presence: true, multiloc: { presence: true }
     post.validates :body_multiloc, presence: true, multiloc: { presence: true, html: true }
-    post.validates :author, presence: true, on: :publication
-    post.validates :author, presence: true, if: :author_id_changed?
+    post.validates :author, presence: true, on: :publication, unless: :anonymous?
+    post.validates :author, presence: true, if: :author_id_changed?, unless: :anonymous?
```

Comment:
I don't think "if" and "unless" can be used together

Reply:
Seemed to work for the tests I was doing.

Reply:
But agree it looks clunky, would be better to put the validation into a single function I think

Reply:
We recently had a similar issue https://github.com/CitizenLabDotCo/citizenlab/pull/4766/files. That's why I strongly suspect the `unless: :draft` condition is no longer working here.

Reply:
But indeed `if` and `unless` seem to work fine together.